src/D2L.CodeStyle.Analyzers/Language/ImmutableGenericDeclarationAnalyzer.cs

```diff
@@ -64,6 +67,15 @@ KnownImmutableTypes knownImmutableTypes
 					continue;
 				}
 
+				var typeName = typeSymbol.GetFullTypeName();
+				if( typeName == "System.Nullable" ) {

```

Comment:
We do it this way in the KnownImmutableTypes, and it's the only way I could find that would consistently identify when System.Nullable was in use.  (Strangely, GetTypeFromMetadataName was *not* working)

Reply:
(To be clear, I mean matching types by name.)